SplitTests/Integration/Api/FlagSetsIntegrationTest.swift

```diff
@@ -0,0 +1,751 @@
+                }
+                self.telemetryStatsExp?.fulfill()
+                return TestDispatcherResponse(code: 200)
+
+            default:
+                return TestDispatcherResponse(code: 500)
+            }
+        }
+    }
+
+    private func buildPollingTestDispatcher() -> HttpClientTestDispatcher {
+        return { request in
+            switch request.url.absoluteString {
+            case let(urlString) where urlString.contains("splitChanges"):
+                if self.splitChangesHit < 3 {
+                    let change = self.pollingFlagSetsHits[self.splitChangesHit]
+                    if self.splitChangesHit > 0 {
+                        self.pollingExps![self.splitChangesHit - 1].fulfill()
+                    }
+                    self.splitChangesHit+=1
+                    return TestDispatcherResponse(code: 200, data: change)
+                }
+                self.splitChangesHit+=1
+                return TestDispatcherResponse(code: 200, data: IntegrationHelper.emptySplitChanges(since: 99999, till: 99999).dataBytes)
+
+            case let(urlString) where urlString.contains("mySegments"):
+                return TestDispatcherResponse(code: 200, data: Data(self.mySegmentsJson.utf8))
+
+            case let(urlString) where urlString.contains("auth"):
+                return TestDispatcherResponse(code: 200, data: Data(IntegrationHelper.dummySseResponse().utf8))
+
+            default:
+                return TestDispatcherResponse(code: 200)
+            }
+        }
+    }
+
+    private func buildStreamingHandler() -> TestStreamResponseBindingHandler {
+        return { request in
+            self.streamingHelper.streamingBinding = TestStreamResponseBinding.createFor(request: request, code: 200)
+            self.sseExp?.fulfill()
+            return self.streamingHelper.streamingBinding!
+        }
+    }
+
+    func testInitialQuerystring() throws {
+
+        let syncConfig = SyncConfig.builder()
+                   .addSplitFilter(SplitFilter.bySet(["set_x", "set_x", "set_3"]))
+                   .addSplitFilter(SplitFilter.bySet(["set_2", "set_3", "set_ww", "invalid+"]))
+                   .build()
+
+        try bodyTest(syncConfig: syncConfig)
+
+        XCTAssertEqual("since=-1&sets=set_2,set_3,set_ww,set_x", querystring)
+    }
+
+    func testTotalAndInvalidFlagSetsTelemetry() throws {
+
+        let syncConfig = SyncConfig.builder()
+                   .addSplitFilter(SplitFilter.bySet(["a", "_b", "a", "a", "c"]))
+                   .addSplitFilter(SplitFilter.bySet(["d", "_d"]))
+                   .build()
+
+        telemetryConfigExp = XCTestExpectation()
+        try bodyTest(syncConfig: syncConfig, telemetryEnabled: true)
+
+        wait(for: [telemetryConfigExp!], timeout: 3)
+
+        XCTAssertEqual("since=-1&sets=a,c,d", querystring)
+        XCTAssertEqual(5, telemetryConfigSent?.flagSetsTotal ?? -1)
+        XCTAssertEqual(2, telemetryConfigSent?.flagSetsInvalid ?? -1)
```

Comment:
Total count should be 7 (`a`, `_b`, `a`, `a`, `c`, `d`, `_d`) & invalid 4 (`_b`, `a`, `a`, `_d`).
```suggestion
        XCTAssertEqual(7, telemetryConfigSent?.flagSetsTotal ?? -1)
        XCTAssertEqual(4, telemetryConfigSent?.flagSetsInvalid ?? -1)
```